### PR TITLE
Document TestableMessageSession

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: Snippets/Testing/Testing_7/MessageSession/MessageSessionTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        var someThingThatUseTheMessageSession = new SomeThingThatUseTheMessageSession(testableSession);
+
+        await someThingThatUseTheMessageSession.DoSomething();
+
+        Assert.AreEqual(1, testableSession.SentMessages.Length);
+        Assert.IsInstanceOf<MyResponse>(testableSession.SentMessages[0].Message);
+    }
+    #endregion
+
+    class SomeThingThatUseTheMessageSession
```
Comment (timbussmann) [inline Snippets/Testing/Testing_7/MessageSession/MessageSessionTests.cs]: should that be `SomethingThatUsesTheMessageSession` ?

File: Snippets/Testing/Testing_6/MessageSession/MessageSessionTests.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        var somethingThatUseTheMessageSession = new SomethingThatUseTheMessageSession(testableSession);
+
+        await somethingThatUseTheMessageSession.DoSomething();
+
+        Assert.AreEqual(1, testableSession.SentMessages.Length);
+        Assert.IsInstanceOf<MyResponse>(testableSession.SentMessages[0].Message);
+    }
+    #endregion
+
+    class SomethingThatUseTheMessageSession
```
Comment (timbussmann) [inline Snippets/Testing/Testing_6/MessageSession/MessageSessionTests.cs]: should be `Uses`?

Comment (andreasohlund) [reply Snippets/Testing/Testing_6/MessageSession/MessageSessionTests.cs]: Think both could work but no strong opinion?

Comment (bording) [reply Snippets/Testing/Testing_6/MessageSession/MessageSessionTests.cs]: @timbussmann is correct, should be "uses"

File: Snippets/Testing/Testing_7/MessageSession/MessageSessionTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        var somethingThatUseTheMessageSession = new SomethingThatUseTheMessageSession(testableSession);
+
+        await somethingThatUseTheMessageSession.DoSomething();
+
+        Assert.AreEqual(1, testableSession.SentMessages.Length);
+        Assert.IsInstanceOf<MyResponse>(testableSession.SentMessages[0].Message);
+    }
+    #endregion
+
+    class SomethingThatUseTheMessageSession
```
Comment (bording) [inline Snippets/Testing/Testing_7/MessageSession/MessageSessionTests.cs]: Same change here

File: nservicebus/testing/index_content_testing_[6,).partial.md [hunk 4]
```diff
@@ -22,7 +22,22 @@ This test verifies that a `Reply` occurred:
...
+| Property | Description |
+|---------------------|----------------------------------------------------------------------------------|
+| `SentMessages` | A list of all messages sent by `session.Send()`. |
+| `PublishedMessages` | A list of all messages published by `session.Publish()`. |
+| `Subscriptions` | A list of all message types explicitly subscribed to using `session.Subscribe()`. |
+| `Unsubscriptions` | A list of all message types explicitly unsubscribed to using `session.Unsubscribe()`. |
+
+### Example
+
+The following code shows how to verify that a message was `Sent` using the `IMessageSession`.
```
Comment (bording) [inline nservicebus/testing/index_content_testing_[6,).partial.md]: Should probably be "was `Sent` using `IMessageSession`", drop the "the" in there.

File: nservicebus/messaging/send-a-message.md [hunk 5]
```diff
@@ -12,20 +12,20 @@ related:
...
+* Sending a command when a HTML form is submitted in an ASP.NET application.
+* Publishing an event when the user clicks a button on a GUI application (see [Publish and Handle an Event](publish-subscribe/publish-handle-event.md)).
 
 To send a message directly from the endpoint:
 
 snippet: BasicSend
 
 partial: batch-sends-performance
 
+Unit testing sends is supported by [the `NServiceBus.Testing` library](/nservicebus/testing/#testing-logging-behavior).
```
Comment (timbussmann) [inline nservicebus/messaging/send-a-message.md]: should that link to the logging section?

**Pull request merged**